hadoop-hdds/docs/content/design/ozone-sts.md

```diff
@@ -0,0 +1,191 @@
+---
+title: AWS STS Design for Ozone S3
+summary: STS Support in Ozone
+date: 2025-10-30
+jira: HDDS-13323
+status: implementing
+author: Ren Koike, Fabian Morgan
+---
+<!--
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
+-->
+
+# AWS STS Design for Ozone S3
+
+# 1. Introduction
+
+S3 credentials used to communicate with Ozone S3 APIs are based on a Kerberos identity.
+
+Historically, the Ozone community has had interest in a REST API capable of programmatically generating 
+temporary S3 credentials.
+
+Amazon AWS has the [Security Token Service (STS)](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html) which 
+provides the ability to generate short-lived access to resources.
+
+The primary scope of this document is to detail the initial implementation of STS within the Ozone ecosystem.
+
+# 2. Why Use STS Tokens?
+
+Providing short-lived access to various resources in Ozone is useful in scenarios such as Data Lake
+solutions that want to aggregate data across multiple cloud providers.
+
+# 3. How Ozone STS Works
+
+The initial implementation of Ozone STS supports only the [AssumeRole](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html)
```

Comment:
Thanks @fmorg-git for preparing the doc.  

The AssumeRole has a few input parameters, some are optional.  Are they all supported? If not, could you clarify what parameters are supported, and what are not.